docs/processes/quality-assurance/testing-and-dev.md

```diff
@@ -0,0 +1,200 @@
+
+- [Test-Driven Development](#test-driven-development)
+  - [Writing Appropriate Tests](#determining-appropriate-tests)
+  - [Writing a Failing Test Case](#writing-a-failing-test-case)
+  - [Writing Code to Pass a Test](#writing-code-to-pass-your-test)
+  - [Refactoring Code](#refactoring-code)
+- [Epics and User Stories](#epics-and-user-stories)
+  - [Epics](#epics)
+  - [User Stories](#user-stories)
+- [Testing Templates](#testing-templates)
+
+Current Areas under construction:
+
+- [Refactoring Intro](#tbc-refactoring-intro)
+- [Waiting epic Template link](#tbc-template-link)
+- [Testing tools expansion](#tbc-testing-tools)
+
+# Testing and Development
+
+Testing should be built into the software development process.
+
+Thoth-Tech advocates for a [Test Driven Development](#test-driven-development) approach.
+
+But you might be wondering, before jumping into coding and planning tests for a new feature, _where do you begin_?
+
+## Feature development – where do I start?
+
+- Create [epics](#epics) for each feature for planned development
+- Identify [user stories](#user-stories) of the feature
+- Consider how best to mitigate risk or look for alternate solutions
+- Determine functional and non-functional requirements
+- Determine acceptance criteria
+  - What are the metrics for success?
+  - What's in/out of scope?
+
+For each product/feature, using the required specifications, flesh out the expected behaviour for different use case scenarios. A collaborative approach helps to make sure you are defining the right problem and solution; be sure to seek feedback from the team during this process.
+
+Armed with your acceptance criteria and use cases, you are ready to create the test scenarios that drive a [test driven development](#test-driven-development) approach.
+
+You will also need to consider _how_ you are going to test your scenarios as part of this process – considering matters such as what tools will you use, what platforms tests will be performed on, any release processes, and what deliverables will you need to produce. This should be considered as part of your Test Strategy – a deliverable which should be produced as part of the QA process. Other documents that may be produced as part of the QA process include a Test Plan and Bug Reports.
+
+Templates have been provided for sample [Test Strategy](templates/test-strategy-template.md), [Test Plan](templates/test-plan-template.md) and [Bug Report](templates/bug-report-template.md) documents.
+
+While manual testing may be conducted initially, moving towards automated tests allow for better efficiency, ease of code checks and is the preferred method to allow automatic test incorporation into the deployment pipeline. Testing and processes, like many other in the Agile model, are not static, and can (and should) be improved on successive iterations. This may include iterating existing processes for testing release as well as automating more functionality.
```

Comment:
```suggestion
While manual testing may be conducted initially, moving towards automated tests allows for better efficiency, ease of code checks and is the preferred method to allow automatic test incorporation into the deployment pipeline. Testing and processes, like many other in the Agile model, are not static, and can (and should) be improved on successive iterations. This may include iterating existing processes for testing releases as well as automating more functionality.
```